src/main/java/seedu/address/model/request/Request.java

```diff
@@ -164,7 +165,7 @@ public boolean equals(Object other) {
             && (otherRequest.phone.equals(this.phone))
             && (otherRequest.nric.equals(this.nric))
             && (otherRequest.getConditions().equals(this.conditions))
-            && otherRequest.getHealthStaff().equals(this.healthWorker)
+            && (Objects.equals(this.healthWorker, otherRequest.getHealthStaff()))
             && (otherRequest.getRequestStatus().equals(this.requestStatus));
```

Comment:
whats the bug here though?

Reply:
I think the getHealthStaff returns a ```String``` but the healthWorker is a ```HealthWorker``` type.

Reply:
`healthWorker` field may not always have a value. If either `other` or `this` has a null `healthWorker` value,  then a `NullPointerException` will be thrown. To avoid this, `Objects.equals()` method is used instead so that if either `healthWorker` has a null value, false is automatically returned.